sympy/core/function.py

```diff
@@ -1707,6 +1707,10 @@ def free_symbols(self):
             ret.update(count.free_symbols)
         return ret
 
+    @property
+    def kind(self):
+        return self.args[0].kind
```

Comment:
It's generally better to avoid accessing `self.args` by index and let specific accessors do that. This case would be `return self.expr.kind`.